plaso/parsers/plist_plugins/ios_identityservices.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""Plist parser plugin for iOS identity services status cache files."""
+
+from dfdatetime import cocoa_time as dfdatetime_cocoa_time
+
+from plaso.containers import events
+from plaso.parsers import plist
+from plaso.parsers.plist_plugins import interface
+
+
+class IOSIdstatusacheEventData(events.EventData):
+  """iOS identity services status cache event data.
+
+  Attributes:
+    apple_identifier (str): type and value of the identifier.
+    lookup_time (dfdatetime.DateTimeValues): date and time of the lookup.
+    process_name (str)" name of the process that looked up an identifier.
+  """
+
+  DATA_TYPE = 'ios:idstatuscache:lookup'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(IOSIdstatusacheEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.apple_identifier = None
+    self.lookup_time = None
+    self.process_name = None
+
+
+class IOSIdstatusachePlistPlugin(interface.PlistPlugin):
+  """Plist parser plugin for identity services status cache files.
+  
+  Identity services status cache plist files are typically named:
+  com.apple.identityservices.idstatuscache.plist
+  """
+
+  NAME = 'io_identityservices_idstatuscache'
+  DATA_FORMAT = 'Idstatuscache plist file'
+
+  PLIST_PATH_FILTERS = frozenset([interface.PlistPathFilter(
+      'com.apple.identityservices.idstatuscache.plist')])
+
+  # pylint: disable=arguments-differ
+  def _ParsePlist(
+      self, parser_mediator, match=None, top_level=None, **unused_kwargs):
+    """Extracts identity services status cache information from the plist.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      match (Optional[dict[str: object]]): keys extracted from PLIST_KEYS.
+      top_level (Optional[dict[str, object]]): plist top-level item.
+    """
+    for _, process_name, process_values in self._RecurseKey(top_level, depth=1):
+      if process_name == 'CacheVersion':
+        continue
+
+      for apple_identifier, apple_identifier_values in process_values.items():
+        event_data = IOSIdstatusacheEventData()
+        event_data.process_name = process_name
+        event_data.apple_identifier = apple_identifier
+        timestamp = apple_identifier_values.get('LookupDate')
```

Comment:
What if this returns None?

Reply:
My apologies, I understood from your previous comment that a timestamp of None was acceptable and that the event would just be dropped by the parser_mediator.

If that is not the case, I should proactively pass the lookup events without a timestamp.

Reply:
so `event_data.lookup_time` can be None but timestamp cannot, otherwise it'll set lookup_time to an empty CocoaTime object (which is not None)

Reply:
I've made some changes, on the assumption that 0 is a valid timestamp.